api/src/main/java/org/openmrs/module/bedmanagement/BedLayoutWithDetails.java

```diff
@@ -48,12 +48,14 @@ public BedLayout convertToBedLayout() {
         bedLayout.setRowNumber(this.rowNumber);
         bedLayout.setColumnNumber(this.columnNumber);
         bedLayout.setLocation(this.location);
-        bedLayout.setBedNumber(this.bed.getBedNumber());
-        bedLayout.setBedId(this.bed.getId());
-        bedLayout.setStatus(this.bed.getStatus());
-        bedLayout.setBedType(this.bed.getBedType());
-        bedLayout.setBedTagMaps(this.bed.getBedTagMap());
-        setPatientInfo(bedLayout);
+        bedLayout.setBedNumber(this.bed != null ? this.bed.getBedNumber() : null);
```

Comment:
Does it really make sense to keep going if `this.bed == null`? Or should we just not throw a runtime exception?